controller_manager/include/controller_manager/loaned_command_interface.hpp

```diff
@@ -0,0 +1,68 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CONTROLLER_MANAGER__LOANED_COMMAND_INTERFACE_HPP_
+#define CONTROLLER_MANAGER__LOANED_COMMAND_INTERFACE_HPP_
+
+#include <utility>
+
+#include "hardware_interface/handle.hpp"
+
+namespace controller_manager
+{
+
+class LoanedCommandInterface
+{
+public:
+  using Deleter = std::function<void (void)>;
+
+  explicit LoanedCommandInterface(hardware_interface::CommandInterface & command_interface)
+  : LoanedCommandInterface(command_interface, nullptr)
```

Comment:
I'd opt for enforcing defining a deleter, even if one wants a nullptr, let them write it explicitly. For command interfaces the normal case is to use a deleter to release the resource.

Reply:
I've ticketed it in #231